docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+ - extensions and ``must_understand = True`` might be too restrictive. Work a
+   We propose to develop a draft implementation with extensions and
+   see how far we can go. A possible list of extensions to include:
+
+    - Boolean
+    - Complex
+    - Datetime
+    - Named dimensions
+    - Awkward arrays
+
+   See https://github.com/zarr-developers/zarr-specs/issues/89 for discussion on
+   the topic.
+
+  - Node name case sensitivity: The node name is now case sensitive. This may
+    make store implementation more complicated as some backends might not be
+    (like some specific filesystem / object store), and we may want to
+    recommend a standard escaping mechanism in those cases.
+    https://github.com/zarr-developers/zarr-specs/issues/57
+
+  - Node name character set: We
+    solicit feedback on whether store implementation should support full unicode.
+    https://github.com/zarr-developers/zarr-specs/issues/56
+
+  - Should named dimensions be part of the core metadata spec?
+    https://github.com/zarr-developers/zarr-specs/issues/73
+
+
+Document conventions
+====================
+
+Conformance requirements are expressed with a combination of
+descriptive assertions and [RFC2119]_ terminology. The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative
+parts of this document are to be interpreted as described in
+[RFC2119]_. However, for readability, these words do not appear in all
+uppercase letters in this specification.
+
+All of the text of this specification is normative except sections
+explicitly marked as non-normative, examples, and notes. Examples in
+this specification are introduced with the words "for example".
+
+
+Concepts and terminology
+========================
+
+This section introduces and defines some key terms and explains the
+conceptual model underpinning the Zarr format.
+
+.. _hierarchy:
+
+*Hierarchy*
+
+    A Zarr hierarchy is a tree structure, where each node in the tree
+    is either a group_ or an array_. Group nodes may have children but
+    array nodes may not. All nodes in a hierarchy have a name_ and a
+    path_.
+
+.. _group:
+.. _groups:
+
+*Group*
+
+    A group is a node in a hierarchy_ that may have child nodes.
+
+.. _array:
+.. _arrays:
+
+*Array*
+
+    An array is a node in a hierarchy_. An array is a data structure
+    with zero or more dimensions_ whose lengths define the shape_ of
+    the array. An array contains zero or more data elements_. All
+    elements_ in an array conform to the same `data type`_. An array
+    may not have child nodes.
+
+.. _name:
+.. _names:
+
+*Name*
+
+    Each node in a hierarchy_ has a name, which is a string of
+    characters with some additional constraints defined in the section
+    on `node names`_ below. Two sibling nodes cannot have the same
+    name. The root node does not have a name.
```

Comment:
Is the name of the root node the empty string `""`, or a non-string like `undefined` / `null`? If the root node is named `""`, then the last sentence of this section is false and should be replaced with something like "The name of the root node is the empty string """.

If the name of the root node is not a string but rather some representation of a missing value like `null` or `undefined`, the first sentence of this section should be updated to "Each node in a hierarchy except the root node has a name, which is a string of characters...." and the last sentence should be made more explicit about the nature of the non-string name of the root node.

Personally I find it preferable to stay in the space of strings and use the empty string as the name of the root node.

Reply:
Does this concern affect the specification itself, or is it purely an editorial issue?

Reply:
it concerns the specification insofar as in one case, the type of node names is `string`, and in the other case its `Union[string, null]`

Reply:
Sorry, I meant, does it affect the stored representation, or is it just a matter of how the spec itself is worded?  I don't think  in the current spec the node name itself is ever used in the stored representation, except for forming keys into the underlying store.

Reply:
Fixed in #74e1667 of PR #171.